pyveda/vedaset/stream/vedastream.py

```diff
@@ -110,12 +110,20 @@ def exhausted(self, val):
 
     @property
     def images(self):
-        _, imgs = zip(*self._vset._buf)
+        try:
+            _, imgs = zip(*self._vset._buf)
+        except ValueError:
+            time.sleep(.5)
+            return self.images
         return BufferedVariableArray(imgs)
 
     @property
     def labels(self):
-        lbls, _ = zip(*self._vset._buf)
+        try:
+            lbls, _ = zip(*self._vset._buf)
+        except ValueError:
+            time.sleep(.5)
+            return self.labels
         return BufferedVariableArray(lbls)
```

Comment:
same, this never happens?

Reply:
initialize_buffer is already being called and I dont think it is blocking. Gets called on start_consumer

Reply:
the issue was that after calling start_consumer immediate access throwing errors. So possibly just needs to be added there.

Reply:
ahh right `f.result()` should block right?

Reply:
initialize_buffer does block, yes

Reply:
so if that wasn't working it means something else is wrong, likely upstream, so sleeping recursively on the accessor methods is not only potentially v bad, but any sleep is unnecessary there anyway